src/drawers/cMapDrawer.cpp

```diff
@@ -148,8 +148,8 @@ void cMapDrawer::drawTerrain()
             }
 
             // draw Smudge if necessary
-            if (cell->smudgetype > -1 && cell->smudgetile > -1) {
-                const cRectangle src_pos = {cell->smudgetile * 32, cell->smudgetype * 32,32, 32};
+            if (cell->smudgetype.has_value() && cell->smudgetile > -1) {
+                const cRectangle src_pos = {cell->smudgetile * 32, static_cast<int>(cell->smudgetype.value()) * 32,32, 32};
```

Comment:
Since I work with `Optional` in Java code bases, I wonder if we can take this a step further. Will experiment.